extensions/GOOGLE/SPV_GOOGLE_user_type.asciidoc

```diff
@@ -0,0 +1,178 @@
+SPV_GOOGLE_user_type
+====================
+
+Name Strings
+------------
+
+SPV_GOOGLE_user_type
+
+Contact
+-------
+
+See *Issues* list in the Khronos SPIRV-Headers repository:
+https://github.com/KhronosGroup/SPIRV-Headers
+
+Contributors
+------------
+
+- Hai Nguyen, Google
+- David Neto, Google
+- Ehsan Nasiri, Google
+
+Status
+------
+- Draft
+
+Version
+-------
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | 2019-04-22
+| Initial            | 1
+|========================================
+
+Dependencies
+------------
+
+This extension is written against the SPIR-V Specification,
+Version 1.2 Revision 4, Unified.
+
+This extension requires SPIR-V 1.0.
+
+This extension also subsumes the SPV_GOOGLE_decorate_string extension.
```

Comment:
This sounds like this is a superset of SPV_GOOGLE_decorate_string, the latter no longer being needed.

But, do you mean it just includes it, so that it does not have to be independently declared?

> Use of this extension implicitly includes SPV_GOOGLE_decorate_string, with no declaration needed for SPV_GOOGLE_decorate_string.

That is when using SPV_GOOGLE_user_type, if one uses a feature of SPV_GOOGLE_decorate_string are they still using SPV_GOOGLE_decorate_string (minus declaration) or are they only using SPV_GOOGLE_user_type?

Reply:
> do you mean it just includes it, so that it does not have to be independently declared?

Yes, that's my understanding. Similar to how `SPV_GOOGLE_hlsl_functionality1` subsumes `SPV_GOOGLE_decorate_string`. [[source]](https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/GOOGLE/SPV_GOOGLE_hlsl_functionality1.asciidoc#dependencies)

Reply:
I think we should be saying something like "implicitly includes" (this one automatically includes the other one) rather than "subsumes" (this one is  bigger than the other, making the other one not necessary).

Reply:
SGTM